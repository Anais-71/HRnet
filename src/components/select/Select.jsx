import React from 'react'
import './select.css'

const Select = ({
  options,
  label,
  value,
  onChange,
  name,
  valueField,
  labelField,
}) => {
  return (
    <div className="select-container">
      <label htmlFor={name}>{label}</label>
      <select
        className="select"
        id={name}
        value={value}
        onChange={onChange}
        name={name}
      >
        <option value="">Please select</option>
        {options.map((option) => (
          <option
            className="select__option"
            key={option[valueField]}
            value={option[valueField]}
          >
            {option[labelField]}
          </option>
        ))}
      </select>
    </div>
  )
}

export default Select
