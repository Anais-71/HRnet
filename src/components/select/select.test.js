import React, { useState } from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import Select from './Select' // Ensure the import path is correct

const mockOptions = [
  { id: 1, label: 'Option 1' },
  { id: 2, label: 'Option 2' },
]

/**
 * Test suite for the Select component.
 */
describe('Select Component', () => {
  /**
   * Test to check if the select field is rendered.
   */
  test('Then the select field is rendered', () => {
    render(
      <Select
        options={mockOptions}
        label="Test Select"
        value=""
        onChange={() => {}}
        name="test-select-1" // Unique ID for this test
        valueField="id"
        labelField="label"
      />,
    )

    const select = screen.getByRole('combobox')
    expect(select).toBeInTheDocument()
  })

  /**
   * Test to verify that the onChange function is called when an option is selected.
   */
  test('When an option is selected, Then the onChange function is called', () => {
    const mockOnChange = jest.fn() // Mock of the onChange function
    const TestComponent = () => {
      const [selectedValue, setSelectedValue] = useState('') // State for the selected value

      const handleChange = (event) => {
        setSelectedValue(event.target.value) // Update state with the new value
        mockOnChange(event) // Call the mocked onChange function
      }

      return (
        <Select
          options={mockOptions}
          label="Test Select"
          value={selectedValue} // Pass the state value
          onChange={handleChange} // Pass the event handler
          name="test-select-2" // Unique ID for this test
          valueField="id"
          labelField="label"
        />
      )
    }

    render(<TestComponent />) // Render the test component

    const select = screen.getByRole('combobox')
    fireEvent.change(select, { target: { value: '1' } }) // Select option 1

    expect(mockOnChange).toHaveBeenCalledTimes(1) // Check that the function was called once
    expect(mockOnChange).toHaveBeenCalledWith(expect.any(Object)) // Check that the function was called with an event object
    expect(select.value).toBe('1') // Check that the selected value has been updated
  })

  /**
   * Test to verify that the select displays the correct options.
   */
  test('Then the select displays the correct options', () => {
    render(
      <Select
        options={mockOptions}
        label="Test Select"
        value=""
        onChange={() => {}}
        name="test-select-3" // Unique ID for this test
        valueField="id"
        labelField="label"
      />,
    )

    const select = screen.getByRole('combobox')
    expect(select).toHaveTextContent(' ') // Check that the default option is present
    expect(select).toHaveTextContent('Option 1') // Check that option 1 is present
    expect(select).toHaveTextContent('Option 2') // Check that option 2 is present
  })

  /**
   * Snapshot test for the Select component.
   */
  test('Select component matches snapshot', () => {
    const { asFragment } = render(
      <Select
        options={mockOptions}
        label="Test Select"
        value=""
        onChange={() => {}}
        name="test-select-snapshot" // Unique ID for this test
        valueField="id"
        labelField="label"
      />,
    )
    expect(asFragment()).toMatchSnapshot() // Check that the component matches the snapshot
  })
})
