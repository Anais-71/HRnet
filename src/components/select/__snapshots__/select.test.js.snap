// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component Select component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="select-container"
  >
    <label
      for="test-select-snapshot"
    >
      Test Select
    </label>
    <select
      class="select"
      id="test-select-snapshot"
      name="test-select-snapshot"
    >
      <option
        class="select__placeholder"
        value=""
      />
      <option
        class="select__option"
        value="1"
      >
        Option 1
      </option>
      <option
        class="select__option"
        value="2"
      >
        Option 2
      </option>
    </select>
  </div>
</DocumentFragment>
`;
